src/components/dialogs/network-modifications/generator/generator-dialog.type.ts

```diff
@@ -0,0 +1,251 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {
+    ACTIVE_POWER_SET_POINT,
+    ADDITIONAL_PROPERTIES,
+    BUS_OR_BUSBAR_SECTION,
+    CONNECTED,
+    CONNECTION_DIRECTION,
+    CONNECTION_NAME,
+    CONNECTION_POSITION,
+    CONNECTIVITY,
+    DROOP,
+    ENERGY_SOURCE,
+    EQUIPMENT,
+    EQUIPMENT_ID,
+    EQUIPMENT_NAME,
+    FORCED_OUTAGE_RATE,
+    FREQUENCY_REGULATION,
+    ID,
+    MARGINAL_COST,
+    MAX_Q,
+    MAXIMUM_ACTIVE_POWER,
+    MAXIMUM_REACTIVE_POWER,
+    MIN_Q,
+    MINIMUM_ACTIVE_POWER,
+    MINIMUM_REACTIVE_POWER,
+    NAME,
+    NOMINAL_VOLTAGE,
+    P,
+    PLANNED_ACTIVE_POWER_SET_POINT,
+    PLANNED_OUTAGE_RATE,
+    Q_PERCENT,
+    RATED_NOMINAL_POWER,
+    REACTIVE_CAPABILITY_CURVE_CHOICE,
+    REACTIVE_CAPABILITY_CURVE_TABLE,
+    REACTIVE_LIMITS,
+    REACTIVE_POWER_SET_POINT,
+    SUBSTATION_ID,
+    TOPOLOGY_KIND,
+    TRANSFORMER_REACTANCE,
+    TRANSIENT_REACTANCE,
+    TYPE,
+    VOLTAGE_LEVEL,
+    VOLTAGE_REGULATION,
+    VOLTAGE_REGULATION_TYPE,
+    VOLTAGE_SET_POINT,
+} from '../../../utils/field-constants';
+import { EquipmentType } from '@gridsuite/commons-ui';
+import { Property } from '../common/properties/property-utils';
+import { ConnectablePositionFormInfos } from '../../connectivity/connectivity.type';
+
+export type ValueWrapper<T> = { value: T };
```

Comment:
move to network-modification-types.ts and define as same name in the back if possible, if so complicated we can typing op as string

```suggestion
export enum OperationType {
    SET = 'SET',
    UNSET = 'UNSET',
}
export type AttributeModification<T> = { 
     value: T;
     op: OperationType;
  };
```